cds_ils/patrons/indexer.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2020 CERN.
+#
+# CDS-ILS is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""Patron Indexer for CDS-ILS."""
+
+from invenio_accounts.models import User
+from invenio_app_ils.patrons.indexer import PatronBaseIndexer
+from invenio_app_ils.proxies import current_app_ils
+from invenio_db import db
+from invenio_oauthclient.models import RemoteAccount
+
+from cds_ils.config import LOCAL_ACCOUNTS_INDEXED
+
+
+def reindex_patrons():
+    """Re-index all patrons."""
+    # do not use PatronIndexer class otherwise it will trigger potentially
+    # thousands of tasks to index referenced records
+    indexer = PatronBaseIndexer()
+    Patron = current_app_ils.patron_cls
+    # cannot use bulk operation because Patron is not a real record
+    all_user_ids = db.session.query(User.id).all()
+    indexed_count = 0
+    for (user_id,) in all_user_ids:
+        if (
+            LOCAL_ACCOUNTS_INDEXED
+            or RemoteAccount.query.filter(
+                RemoteAccount.user_id == user_id
+            ).first()
+            is not None
+        ):
```

Comment:
for readability:
```suggestion
     index_local_account = current_app.config['INDEX_LOCAL_ACCOUNTS')
     has_remote_account = RemoteAccount.query.filter_by(user_id=user_id).first()
     if index_local_account or has_remote_account:
```

other: 
in the `or` statement `None` will be evaluated as false, there is no need for additional check `is not None`

another optimisation:
wouldn't it be better if :
```python
if index_local_account:
    all_user_ids = db.session.query(User.id).all()
else:
    all_user_ids = db.session.query(RemoteAccount.user_id).all()
```

in this case in each option you make only one db query not two